File: src/main/java/com/github/magneticflux/aoc/year2015/day1/Problem1.java
package com.github.magneticflux.aoc.year2015.day1;

import java.util.Iterator;
import java.util.Scanner;

/**
 * Created by Mitchell Skaggs on 12/18/15.
 */
public class Problem1 {
    public static void main(String[] args) {
        Iterator<String> input = new Scanner(Problem1.class.getResourceAsStream("day1.txt"));
        String s = input.next();
        long numOpen = s.chars().filter(character -> character == (int) '(').count();
        long numClosed = s.length() - numOpen;
        System.out.println(numOpen - numClosed);
    }
}

File: src/main/kotlin/com/github/magneticflux/aoc/day4/Problem1.kt
package com.github.magneticflux.aoc.day4

import com.github.magneticflux.aoc.input
import java.util.*

/**
 * Created by Mitchell Skaggs on 12/1/2017.
 */
object Problem1 {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(input())
        input.useDelimiter("\n")

        val count = input.iterator().asSequence().filter {
            val list = it.split(" ")
            val set = list.toSet()
            set.size == list.size
        }.count()

        println("Count= $count")
    }
}